src/GitTools.jl

```diff
@@ -127,4 +127,34 @@ function fetch(repo::LibGit2.GitRepo, remoteurl=nothing; header=nothing, kwargs.
     end
 end
 
+function validate_git_url(url::AbstractString)
+    if ispath(url)
+        try
+            LibGit2.GitRepo(url)
+        catch err
+            if err isa LibGit2.GitError &&
+                    err.code == LibGit2.Error.ENOTFOUND &&
+                    err.class == LibGit2.Error.Repository
+                Pkg.Types.cmderror("'$(url)' does not point to a valid git repository")
+            else
+                throw(err)
+            end
+        end
+    else
+        try
+            # try to fetch anything just to see if it throws an error
```

Comment:
Won't the extra fetch give extra latency? Can't we put this inside `GitTools.fetch` and `GitTools.clone` to not do a redundant fetch?

Reply:
Ok, that would be a lot better.